app/src/main/java/com/example/dragonx/domain/work/RefreshDataWork.kt

```diff
@@ -0,0 +1,27 @@
+package com.example.dragonx.domain.work
+
+import android.content.Context
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.example.dragonx.domain.Retrofit
+import com.example.dragonx.domain.db.RocketDatabase
+import com.example.dragonx.domain.db.RocketRepository
+import retrofit2.HttpException
+
+
+class RefreshDataWorker(appContext: Context, params: WorkerParameters):
+        CoroutineWorker(appContext, params) {
+    companion object {
+        const val WORK_NAME = "RefreshDataWorker"
+    }
+    override suspend fun doWork(): Result {
+        val database = RocketDatabase.getInstance(applicationContext)
+        val repository = RocketRepository(database.rocketDao(), Retrofit.buildApiService())
+        return try {
+            repository.getAllRockets()
```

Comment:
What exactly you doing in the background worker?

Reply:
I refresh data periodically. Another code you may find in RocketApplication